dask/array/core.py

```diff
@@ -1299,7 +1299,22 @@ def finalize(results):
             return concatenate3(results)
         else:
             results2 = results2[0]
-    return unpack_singleton(results)
+
+    results = unpack_singleton(results)
+    # Single chunk. There is a risk that the result holds a buffer stored in the
+    # graph or on a process-local Worker. Deep copy to make sure that nothing can
+    # accidentally write back to it.
+    try:
+        return results.copy()  # numpy, sparse, scipy.sparse (any version)
+    except AttributeError:
+        pass
+    try:
```

Comment:
Can we cross this bridge when we get there? This is currently not something that we are working or focusing on

Reply:
Removed section.